pkg/utils/resource/update/update.go

```diff
@@ -0,0 +1,72 @@
+package update
+
+import (
+	"context"
+	"reflect"
+
+	cco "github.com/openshift/cloud-credential-operator/pkg/apis/cloudcredential/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift/aws-load-balancer-operator/pkg/utils"
+)
+
+// CredentialsRequest updates the given CredentialsRequest to the desired state if needed.
+// Returns true if the CredentialsRequest was updated, false otherwise.
+func CredentialsRequest(ctx context.Context, client client.Client, current *cco.CredentialsRequest, desired *cco.CredentialsRequest) (bool, error) {
+	if changed, err := IsCredentialsRequestChanged(current, desired); err != nil {
+		return false, err
+	} else if !changed {
+		return false, nil
+	}
+
+	updated := current.DeepCopy()
+	updated.Name = desired.Name
+	updated.Namespace = desired.Namespace
+	updated.Spec = desired.Spec
+	if err := client.Update(ctx, updated); err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+// IsCredentialsRequestChanged compares the current and desired states of a CredentialsRequest.
+// Returns true of current start is different from the desired, false otherwise.
+func IsCredentialsRequestChanged(current, desired *cco.CredentialsRequest) (bool, error) {
+	if current.Name != desired.Name {
+		return true, nil
+	}
+
+	if current.Namespace != desired.Namespace {
+		return true, nil
+	}
+
```

Comment:
Did you consider https://pkg.go.dev/github.com/google/go-cmp/cmp? I think it works nicely for simple comparisons, but not so sure about more advanced. 

Some examples:
- https://github.com/openshift/cluster-ingress-operator/blob/6cd3b6709eff0026ad814921b40c4b02c32fb38c/pkg/operator/controller/ingress/internal_service.go#L116
- https://github.com/openshift/cluster-ingress-operator/blob/d7ffd6e7a9cd89805e273b37fff6e6cca6298677/pkg/operator/controller/gatewayapi/crds.go#L115

Reply:
I didn't want to change the logic as I reintegrated it back to the existing code, so I didn't want any change in the behavior. I think it was considered at the time of the creation. I'm fine with this more verbose approach as it explicitly states what we are interested in and let the rest be defaulted or ignored.

Btw, I'm always a little surprised how wide spread it is in the production code taking into account their note in the package description:
> It is intended to only be used in tests, as performance is not a goal and it may panic if it cannot compare the values.